File: Leetcode/6 - Zigzag Conversion/solution.js
var convert = function (s, numRows) {
    if ((numRows === 1) || (s.length <= numRows)) return s;

    let zigZag = Array.from({ length: numRows }, () => "");
    let goingDown = true;
    let row = 0;

    for (let i = 0; i < s.length; i++) {
        zigZag[row] += s[i];
        if (row === (numRows - 1)) {
            goingDown = false;
        } else if (row === 0) {
            goingDown = true;
        }
        row += (goingDown ? 1 : -1);
    }

    return zigZag.join('');
};

File: Leetcode/6 - Zigzag Conversion/solution.py
def convert(self, s: str, num_rows: int) -> str:
    if num_rows == 1 or len(s) <= num_rows:
        return s

    zigzag: list[str] = ['' for _ in range(num_rows)]
    going_down: bool = True
    row: int = 0

    for char in s:
        zigzag[row] += char
        if row == num_rows - 1:
            going_down = False
        elif row == 0:
            going_down = True
        row += 1 if going_down else -1

    return ''.join(zigzag)